yaml/src/main/java/com/fasterxml/jackson/dataformat/yaml/YAMLGenerator.java

```diff
@@ -517,13 +524,16 @@ public final void writeStartArray() throws IOException
     {
         _verifyValueWrite("start an array");
         _writeContext = _writeContext.createChildArrayContext();
-        FlowStyle style = _outputOptions.getDefaultFlowStyle();
         String yamlTag = _typeId;
         boolean implicit = (yamlTag == null);
         String anchor = _objectId;
         if (anchor != null) {
             _objectId = null;
         }
+        NodeStyle jacksonStyle = _nodeStyleResolver.resolveStyle(
```

Comment:
Would make sense to use `getCurrentName()` bit earlier here, before `createChildArrayContext()`.

Would this work with nested arrays? (might be that it gets `null`).

Also: would make sense to get and pass default style to custom resolver -- but resolver could be passed the context as well (pre-addition of array), to let it access all information, including ancestors. I'll add a note on interface later on.